binary_search/binary_search.cpp

```diff
@@ -0,0 +1,51 @@
+#include <iostream>
+#include <vector>
+using namespace std;
+
+//------------------------------------------------
```

Comment:
Please make the comments in this style - https://github.com/iiitv/algos/blob/master/longest_palindromic_substring/longest_palindromic_substring.cpp.